pkg/kubernetes/kubernetes.go

```diff
@@ -250,19 +258,27 @@ func (c *ClientsCollection) deleteIfFound(ctx context.Context, logger *logrus.En
 func (c *ClientsCollection) DeleteResource(ctx context.Context, logger *logrus.Entry, deletion *Resource) error {
 	logger = logger.WithFields(deletion.LogFields())
 
-	// identify the resource to be deleted either by name or
-	// labels. name AND labels cannot be defined at the same time,
-	// but one of them MUST be defined.
-	if deletion.Name != "" && len(deletion.Labels) > 0 {
-		return fmt.Errorf("only one of 'name' or 'labels' must be specified")
+	// identify the resource to be deleted either by name, selector or labels.
+	// Only one of them must be defined.
+	ids := 0
```

Comment:
nit: what does the variable name `ids` mean? maybe it could be more descriptive?

Reply:
renamed to `resourceIdentifiers`